website/src/views/conceptMap.js

```diff
@@ -73,15 +75,41 @@ export default function ConceptMap() {
         let mounted = true;
         if (mounted) {
             setLoading(true);
-            // Fetch the student progressQueryString
-            apiv2.get(`/students/${selectedStudent}/progressquerystring`).then((res) => {
-                setStudentMastery(res.data);
+            // Fetch the student masterymapping
+            apiv2.get(`/students/${selectedStudent}/masterymapping`).then((res) => {
+                setMasteryMapping(res.data);
                 setLoading(false);
             });
         }
         return () => mounted = false;
     }, [selectedStudent])
 
+    useEffect(() => {
+        async function fetchCMHTML() {
+            const url = `${window.location.origin}/progress`;
+            try {
+                const response = await fetch(url, {
+                    method: "POST",
+                    headers: {
+                        "Content-Type": "application/json"
+                    },
+                    body: JSON.stringify(masteryMapping)
+                });
+                if (!response.ok) {
+                    throw new Error(`HTTP error! Status: ${response.status}`);
+                }
+                const result = await response.text();
+                return result;
+            } catch (error) {
+                console.error("Error:", error);
+                return null;
+            }
+        }
+        fetchCMHTML().then(result => {
+            setConceptMapHTML(result);
+        });
+    })
```

Comment:
This should be done with axios as we've done in other pages for consistency.  Also, you probably do not want your function def to be within the `useEffect` since that will re-render every time your `useEffect` triggers.  You could define it with a `useCallback` or `useMemo`.  Also, where do you define the dependency array for this `useEffect`?  I highly recommend taking a look at [the official documentation](https://react.dev/reference/react/useEffect) as I find it is really good.  You're also welcome to ask me questions directly.

Reply:
Thanks, Connor! Will take a look.

Reply:
Modified accordingly!